matrix-api.h

```diff
@@ -67,6 +71,16 @@ typedef void (*MatrixApiCallback)(MatrixConnectionData *conn,
                                   const char *body,
                                   size_t body_len, const char *content_type);
 
+void room_create_callback(MatrixConnectionData *conn,
```

Comment:
these callbacks should probably be static functions in the place where they are used. They don't fit the public API of matrix-api.{c,h}.